azurelinuxagent/agent.py

```diff
@@ -186,15 +186,15 @@ def parse_args(sys_args):
     verbose = False
     conf_file_path = None
     for a in sys_args:
-        m = re.match("^(?:[-/]*)configuration-path:([\w/\.\-_]+)", a)
-        if not m is None:
+        m = re.match(r"^(?:[-/]*)configuration-path:([\w/\.\-_]+)", a)
+        if m is not None:
             conf_file_path = m.group(1)
             if not os.path.exists(conf_file_path):
                 print("Error: Configuration file {0} does not exist".format(
                         conf_file_path), file=sys.stderr)
                 usage()
                 sys.exit(1)
-        
+
```

Comment:
were you trying to use raw strings for all regular expressions? there's a couple more @ 198, 200 (watch out with \\)

Reply:
Yes, I fixed the ones that the nightly build identified as problematic, since python 3.8 changes regular expression string handling.